testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -443,6 +443,26 @@ def test_gaps(self, universe, Writer, outfile):
                 with pytest.raises(mda.NoDataError):
                     getattr(written_ts, 'velocities')
 
+    def test_data_preservation(self, universe, Writer, outfile):
+
+        with Writer(outfile, universe.atoms.n_atoms, dt=universe.trajectory.dt) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(GRO, outfile)
+
+        assert np.isclose(ts.data['time'], 0.0)
+        assert ts.data['step'] == 0
+        assert np.isclose(ts.data['lambda'], 0.0)
+        assert np.isclose(ts.data['dt'], 100.0)
+
+        # check that the data are identical for each time step
+        for orig_ts, written_ts in zip(universe.trajectory, uw.trajectory):
+            err_msg = ('mismatch between '
+                       'original and written trajectory at '
+                       f'frame {orig_ts.frame} vs {written_ts.frame}')
+            assert_equal(orig_ts.data, written_ts.data, err_msg=err_msg)
```

Comment:
I think you want assert_almost_equal for floats

Reply:
Note that `assert_almost_equal` is being phased out in favour of `assert_allclose` but makes no difference here AFAIK. Im happy either way.

Reply:
Just going to use `assert_allclose`